backend/agentd/session_test.go

```diff
@@ -307,3 +320,64 @@ func TestSessionEntityWatchDeleteAndUnknown(t *testing.T) {
 		t.Fatal("broken session never stopped")
 	}
 }
+
+func TestSessionInvalidEntityClassUpdate(t *testing.T) {
+	wait := make(chan struct{})
+
+	conn := new(mocktransport.MockTransport)
+	// Mock the Receive method by blocking it for 100ms and returns an empty
+	// message so it doesn't block our test for too long
+	conn.On("Receive").After(100*time.Millisecond).Return(&transport.Message{}, nil)
```

Comment:
Does the test still work if you return immediately?

Reply:
Oh yeah absolutely, it's just that the actual implementation is a blocking channel so it would constantly call this mocked method; the 100ms wait is just so it spams a bit less but doesn't delay the test execution too much.